library/alloc/tests/string.rs

```diff
@@ -114,6 +114,43 @@ fn test_from_utf8_lossy() {
     );
 }
 
+#[test]
+fn test_fromutf8error_into_lossy() {
+    fn func(input: &[u8]) -> String {
+        String::from_utf8(input.to_owned()).unwrap_or_else(|e| e.into_utf8_lossy())
+    }
+
+    let xs = b"hello";
+    let ys = "hello".to_owned();
+    assert_eq!(func(xs), ys);
+
+    let xs = "ศไทย中华Việt Nam".as_bytes();
+    let ys = "ศไทย中华Việt Nam".to_owned();
+    assert_eq!(func(xs), ys);
+
+    let xs = b"Hello\xC2 There\xFF Goodbye";
+    assert_eq!(func(xs), String::from("Hello\u{FFFD} There\u{FFFD} Goodbye"));
```

Comment:
nit (you don't have to change this but can if you're touching it anyways and want to): you've used `.to_owned()` above but use `String::from` here, i prefer `.to_owned()` myself so they could be made consistent

Reply:
Addressed.

I just copied the test above but I prefer that too.